datafusion/sql/src/unparser/expr.rs

```diff
@@ -567,6 +568,43 @@ impl Unparser<'_> {
         Ok(ast::Expr::CompoundIdentifier(id))
     }
 
+    fn map_to_sql(&self, args: &[Expr]) -> Result<ast::Expr> {
+        if args.len() != 2 {
+            return internal_err!("map must have exactly 2 arguments");
+        }
+
+        let keys = match self.expr_to_sql(&args[0])? {
+            ast::Expr::Array(Array { elem, .. }) => elem,
+            other => {
+                return internal_err!(
+                    "map expects first argument to be an array, but received: {:?}",
+                    other
+                )
+            }
+        };
```

Comment:
You can also write this in the following way which may be more concise:

```suggestion
        let  ast::Expr::Array(Array { elem: keys, .. })) = self.expr_to_sql(&args[0])? else {
                return internal_err!(
                    "map expects first argument to be an array, but received: {:?}",
                    other
                )
        };
```